tokio/tests/rt_handle_block_on.rs

```diff
@@ -0,0 +1,491 @@
+        let _enter = rt.enter();
+
+        let bind_future = net::TcpListener::bind("127.0.0.1:0");
+
+        rt.shutdown_timeout(Duration::from_secs(1000));
+
+        let err = Handle::current().block_on(bind_future).unwrap_err();
+
+        assert_eq!(err.kind(), std::io::ErrorKind::Other);
+        assert_eq!(
+            err.get_ref().unwrap().to_string(),
+            "A Tokio 1.x context was found, but it is being shutdown.",
+        );
+    }
+
+    #[test]
+    fn udp_socket_bind() {
+        let rt = rt();
+        let _enter = rt.enter();
+
+        Handle::current()
+            .block_on(net::UdpSocket::bind("127.0.0.1:0"))
+            .unwrap();
+    }
+
+    #[test]
+    fn udp_stream_bind_after_shutdown() {
+        let rt = rt();
+        let _enter = rt.enter();
+
+        rt.shutdown_timeout(Duration::from_secs(1000));
+
+        let err = Handle::current()
+            .block_on(net::UdpSocket::bind("127.0.0.1:0"))
+            .unwrap_err();
+
+        assert_eq!(err.kind(), std::io::ErrorKind::Other);
+        assert_eq!(
+            err.get_ref().unwrap().to_string(),
+            "A Tokio 1.x context was found, but it is being shutdown.",
+        );
+    }
+
+    #[test]
+    fn udp_stream_bind_before_shutdown() {
+        let rt = rt();
+        let _enter = rt.enter();
+
+        let bind_future = net::UdpSocket::bind("127.0.0.1:0");
+
+        rt.shutdown_timeout(Duration::from_secs(1000));
+
+        let err = Handle::current().block_on(bind_future).unwrap_err();
+
+        assert_eq!(err.kind(), std::io::ErrorKind::Other);
+        assert_eq!(
+            err.get_ref().unwrap().to_string(),
+            "A Tokio 1.x context was found, but it is being shutdown.",
+        );
+    }
+
+    #[cfg(unix)]
+    #[test]
+    fn unix_listener_bind_after_shutdown() {
+        let rt = rt();
+        let _enter = rt.enter();
+
+        let dir = tempdir().unwrap();
+        let path = dir.path().join("socket");
+
+        rt.shutdown_timeout(Duration::from_secs(1000));
+
+        let err = net::UnixListener::bind(path).unwrap_err();
+
+        assert_eq!(err.kind(), std::io::ErrorKind::Other);
+        assert_eq!(
+            err.get_ref().unwrap().to_string(),
+            "A Tokio 1.x context was found, but it is being shutdown.",
+        );
+    }
+
+    #[cfg(unix)]
+    #[test]
+    fn unix_listener_bind_accept_after_shutdown_1() {
+        let rt = rt();
+        let _enter = rt.enter();
+
+        let dir = tempdir().unwrap();
+        let path = dir.path().join("socket");
+
+        let listener = net::UnixListener::bind(path).unwrap();
+
+        rt.shutdown_timeout(Duration::from_secs(1000));
+
+        // this should not timeout but fail immediately since the runtime has been shutdown
+        let err = Handle::current().block_on(listener.accept()).unwrap_err();
+
+        assert_eq!(err.kind(), std::io::ErrorKind::Other);
+        assert_eq!(err.get_ref().unwrap().to_string(), "reactor gone");
```

Comment:
This error message was already there. I think it makes sense to change it to `A Tokio 1.x context was found, but it is being shutdown.` for consistency. Wondering if that would be breaking change though.